api/src/main/java/io/opencensus/trace/Tracer.java

```diff
@@ -150,6 +151,135 @@ public final Scope withSpan(Span span) {
     return CurrentSpanUtils.withSpan(checkNotNull(span, "span"), false);
   }
 
+  /**
+   * Wraps a {@link Runnable} so that it executes with the {@code Span} as the current {@code Span}.
+   *
+   * <p>Users may consider to use {@link SpanBuilder#startSpanAndRun(Runnable)}.
+   *
+   * <p>Any error will end up as a {@link Status#UNKNOWN}.
+   *
+   * <p>IMPORTANT: Caller must manually propagate the entire {@code io.grpc.Context} when wraps a
+   * {@code Runnable}, see the examples.
+   *
+   * <p>IMPORTANT: Caller must manually end the {@code Span} within the {@code Runnable}, or after
+   * the {@code Runnable} is executed.
+   *
+   * <p>Example with Executor wrapped with {@link io.grpc.Context#currentContextExecutor}:
+   *
+   * <pre><code>
+   * class MyClass {
+   *   private static Tracer tracer = Tracing.getTracer();
+   *   void handleRequest(Executor executor) {
+   *     Span span = new Tracer.spanBuilder("MyRunnableSpan");
+   *     executor.execute(tracer.wrap(span, new Runnable() {
+   *      {@literal @}Override
+   *       public void run() {
+   *         try {
+   *           sendResult();
+   *         } finally {
+   *           span.end();
+   *         }
+   *       }
+   *     }));
+   *   }
+   * }
+   * </code></pre>
+   *
+   * <p>Example without Executor wrapped with {@link io.grpc.Context#currentContextExecutor}:
+   *
+   * <pre><code>
+   * class MyClass {
+   *   private static Tracer tracer = Tracing.getTracer();
+   *   void handleRequest(Executor executor) {
+   *     Span span = new Tracer.spanBuilder("MyRunnableSpan");
+   *     executor.execute(Context.wrap(tracer.wrap(span, new Runnable() {
+   *      {@literal @}Override
+   *       public void run() {
+   *         try {
+   *           sendResult();
+   *         } finally {
+   *           span.end();
+   *         }
+   *       }
+   *     })));
+   *   }
+   * }
+   * </code></pre>
+   *
+   * @param span the {@code Span} to be set as current.
+   * @param runnable the {@code Runnable} to wrap in the {@code Span}.
+   * @return the wrapped {@code Runnable}.
+   * @since 0.11.0
+   */
+  public final Runnable wrap(Span span, Runnable runnable) {
```

Comment:
this api feels weird as I see a lot of wrap utilities which start and stop the span during the work method (ex callable.call). This wraps, but doesn't start it. I wouldn't use this api

Reply:
I think this comment is based on the misunderstanding of the CurrentSpanUtil privacy. This is the only public "wrap" API.